core/src/main/scala/kafka/admin/AclCommand.scala

```diff
@@ -626,18 +637,22 @@ object AclCommand extends Logging {
     options = parser.parse(args: _*)
 
     def checkArgs(): Unit = {
-      if (options.has(bootstrapServerOpt) && options.has(authorizerOpt))
-        CommandLineUtils.printUsageAndExit(parser, "Only one of --bootstrap-server or --authorizer must be specified")
+      if (options.has(bootstrapServerOpt) && options.has(bootstrapControllerOpt))
+        CommandLineUtils.printUsageAndExit(parser, "Only one of --bootstrap-server or --bootstrap-controller must be specified")
+
+      val hasServerOrController = options.has(bootstrapServerOpt) || options.has(bootstrapControllerOpt)
+      if (hasServerOrController && options.has(authorizerOpt))
+        CommandLineUtils.printUsageAndExit(parser, "The --authorizer option can only be used without --bootstrap-server or --bootstrap-controller")
 
-      if (!options.has(bootstrapServerOpt)) {
+      if (!options.has(bootstrapServerOpt) && !options.has(bootstrapControllerOpt)) {
```

Comment:
`!options.has(bootstrapServerOpt) && !options.has(bootstrapControllerOpt)`  can be replaced by `hasServerOrController`